draft-ietf-gnap-core-protocol.md

```diff
@@ -5045,10 +5049,31 @@ software to look up the details of the pending request. Since this approach requ
 by the client software during the redirection process, clients that are not capable of holding state
 through a redirect should not use redirect-based interaction mechanisms.
 
+## Redirection Status Codes {#security-redirect-status-codes}
+
+As already described in {{I-D.ietf-oauth-security-topics}}, a server should never use the HTTP 307
+status code to redirect a request that potentially contains user credentials. If an HTTP redirect
+is used for such a request, the HTTP status code 303 "See Other" should be used instead.
+
+The status code 307, as defined in the HTTP standard {{RFC7231}}, does not require the user agent
+to rewrite a POST request into a GET request, thus preserving the form data in the body of the POST
+request. In the HTTP standard {{RFC7231}}, only the status code 303 unambiguously enforces
+rewriting the HTTP POST request to an HTTP GET request. For all other status codes, including
+status code 302, user agents are allowed not to rewrite a POST request into a GET request and thus
+to resubmit the body.
+
+The use of status code 307 results in a vulnerability when using the
+[`redirect` interaction finish method](#response-interact-finish). With this method, the AS
+potentially prompts the RO to enter their credentials in a form that is then submitted back to the
+AS (using an HTTP POST request). The AS checks the credentials and, if successful, may directly
+redirect the RO to the client instance's redirect URI. Due to the use of status code 307, the RO's
+user agent now transmits the RO's credentials to the client instance. A malicious client instance
+can then use the obtained credentials to impersonate the RO at the AS.
+
```

Comment:
```suggestion

Redirection away from the initial URL in an interaction session could also leak information found in that
initial URL through the HTTP Referrer header field, which would be sent by the user agent to the redirect
target. To avoid such leakage, a server can first redirect to an internal interstitial page without any identifying
or sensitive information on the URL before processing the request. When the user agent is ultimately 
redirected from this page, no part of the original interaction URL will be found in the Referrer header.

```

Reply:
I'm not sure if it should rather be called "HTTP `Referer` header", since the official name of the header is a misspelling. But I'm also not sure what the conventions are for this. Couldn't the HTTP `Referrer-Policy` header also be used for this?